app/services/status_message_service.rb

```diff
@@ -0,0 +1,86 @@
+class StatusMessageService
+  include Rails.application.routes.url_helpers
+
+  attr_reader :status_message
+
+  def initialize(params, user)
+    @user = user
+    normalize_aspect_ids(params)
+    normalize_public_flag!(params)
+    @destination_aspect_ids = destination_aspect_ids(params)
+    @status_message = user.build_post(:status_message, params[:status_message])
+
+    build_location(params)
+    build_poll(params)
+    build_photos(params)
+    build_services(params)
+  end
+
+  def save
+    @status_message.save
+  end
+
+  def process_status
```

Comment:
I think I'd call it `process_message`, since it's a `StatusMessage`, not a `MessageStatus`. I'm also not sure about returning it and relying on the return value here, I think I'd just require calling the accessor in any case.